controllers/managedocs_controller.go

```diff
@@ -136,6 +136,13 @@ func (r *ManagedOCSReconciler) reconcilePhases() error {
 	// Update the status of the components
 	r.managedOCS.Status.Components = r.updateComponents()
 
+	// Update the readiness of the operator.
+	if r.managedOCS.Status.Components.StorageCluster.State != v1.ComponentReady {
+		utils.SetNotReady()
+	} else {
+		utils.SetReady()
+	}
+
```

Comment:
I would prefer the controller itself will have no relation to the prob, so that future PR could easily move it into its own binary.  

We have reflected the components state on the `ManagedOCS` status in order to provide the prob code a way to query the status without any need for in-memory communication between the controller and the prob code

Reply:
Addressed in new commit.